specifyweb/workbench/upload/auditor.py

```diff
@@ -12,11 +12,12 @@
 class Auditor(NamedTuple):
     collection: Any
     audit_log: Optional[AuditLog]
-
+    skip_create_permission_check: bool = False
     def insert(self, inserted_obj: Any, agent: Union[int, Any], parent_record: Optional[Any]) -> None:
+
         if agent is None:
             logger.warn('WB inserting %s with no createdbyagent. Skipping permissions check.', inserted_obj)
-        else:
+        elif not self.skip_create_permission_check:
             if isinstance(agent, int):
                 # TODO: Optimize this potential w/ just memoization
                 agent_obj = Agent.objects.get(id=agent)
```

Comment:
(optional) 
While we're here, we _could_ do some refactoring of this code. 
I would extract this validation logic into it's own function. 
Perhaps something like 
```py
def validate(self, agent: Union[int, Any], inserted_obj: Any) -> None:
        if agent is None:
            logger.warn('WB inserting %s with no createdbyagent. Skipping permissions check.', inserted_obj)
            return
        
        if self.skip_create_permission_check: return

        if isinstance(agent, int):
            # TODO: Optimize this potential w/ just memoization
            agent_obj = Agent.objects.get(id=agent)
            check_table_permissions(self.collection, agent_obj, inserted_obj, "create")
```

> this code is functionally identical to how it was before. Not sure why the `check_table_permissions(...)` was outside of the if statement which defines `agent_obj` before. If `agent` is not an int, pretty sure it would throw an error stating `agent_obj` was used before it was initialized. If this is intentional, we should throw a more explicit and descriptive error. 

and then `insert(...)` can be simplified to 

```py
def insert(self, inserted_obj: Any, agent: Union[int, Any], parent_record: Optional[Any]) -> None:
        self.validate(agent, inserted_obj)

        if self.audit_log is not None:
            self.audit_log.insert(inserted_obj, agent, parent_record)
```

Reply:
This can be opened as a new issue if you'd like!